root/static/scripts/edit/utility/guessFeat.js

```diff
@@ -21,218 +21,248 @@ import {
+           * is also an artist name.
+           */
+          const expandedCredits = expandCredit(
+            m.name, artists, isProbablyClassical,
+          );
+
+          if (_.some(
+            expandedCredits, c => c.similarity >= MIN_NAME_SIMILARITY,
+          )) {
+            credits = credits.concat(expandedCredits);
+          } else {
+            name += pair[0] + m.name + pair[1];
+          }
         }
 
-        return {name: clean(name), joinPhrase: joinPhrase, artistCredit: credits};
-    }, {name: str, joinPhrase: '', artistCredit: []});
+        joinPhrase = joinPhrase || m.joinPhrase;
+        credits = credits.concat(m.artistCredit);
+        remainder = b.post;
+      } else {
+        name += remainder;
+        break;
+      }
+    }
+
+    return {name: clean(name), joinPhrase: joinPhrase, artistCredit: credits};
+  }, {name: str, joinPhrase: '', artistCredit: []});
 }
 
-function extractFeatCredits(str, artists, isProbablyClassical, allowEmptyName) {
-    var m1 = extractBracketedFeatCredits(str, artists, isProbablyClassical);
+function extractFeatCredits(
+  str, artists, isProbablyClassical, allowEmptyName,
+) {
+  const m1 = extractBracketedFeatCredits(str, artists, isProbablyClassical);
 
-    if (!m1.name && !allowEmptyName) {
-        return {name: str, joinPhrase: '', artistCredit: []};
-    }
+  if (!m1.name && !allowEmptyName) {
+    return {name: str, joinPhrase: '', artistCredit: []};
+  }
 
-    var m2 = extractNonBracketedFeatCredits(m1.name, artists, isProbablyClassical);
+  const m2 = extractNonBracketedFeatCredits(
+    m1.name, artists, isProbablyClassical,
+  );
 
-    if (!m2.name && !allowEmptyName) {
-        return m1;
-    }
+  if (!m2.name && !allowEmptyName) {
+    return m1;
+  }
 
-    return {name: m2.name, joinPhrase: m2.joinPhrase || m1.joinPhrase, artistCredit: m2.artistCredit.concat(m1.artistCredit)};
+  return {
+    name: m2.name,
+    joinPhrase: m2.joinPhrase || m1.joinPhrase,
+    artistCredit: m2.artistCredit.concat(m1.artistCredit),
+  };
 }
 
 function cleanCredit(name, isProbablyClassical) {
-    // remove classical roles
-    return isProbablyClassical ? name.replace(/^[a-z]+: (.+)$/, '$1') : name;
+  // remove classical roles
+  return isProbablyClassical ? name.replace(/^[a-z]+: (.+)$/, '$1') : name;
 }
 
 function bestArtistMatch(artists, name) {
-    return _(artists)
-        .map(function (a) {
-            var similarity = getSimilarity(name, a.name);
-            if (similarity >= MIN_NAME_SIMILARITY) {
-                return {similarity: similarity, artist: a, name: name};
-            }
-            return null;
-        })
-        .compact()
-        .sortBy('similarity')
-        .reverse()
-        .head();
+  return _(artists)
+    .map(function (a) {
+      const similarity = getSimilarity(name, a.name);
+      if (similarity >= MIN_NAME_SIMILARITY) {
+        return {similarity: similarity, artist: a, name: name};
+      }
+      return false;
```

Comment:
`null` makes more sense here to me, how it was before. You can also remove `consistent-return` from the commit message since those were already fixed by 6e3446d27e9